### PR TITLE
[DO NOT MERGE] AF-2833 : Conscious language initiative kie-wb-common and kie-wb-playground

### PR DESCRIPTION
**Pull request by akumar074**
**JIRA**: 
[AF-2833](https://issues.redhat.com/browse/AF-2833)
[AF-2829](https://issues.redhat.com/browse/AF-2829)

**referenced Pull Requests**: 
* https://github.com/kiegroup/kie-soup/pull/219
* https://github.com/kiegroup/appformer/pull/1178
* https://github.com/kiegroup/kie-wb-playground/pull/117
* https://github.com/kiegroup/optaplanner-wb/pull/401
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1708
* https://github.com/kiegroup/drools-wb/pull/1509
* https://github.com/kiegroup/jbpm-designer/pull/914
* https://github.com/kiegroup/jbpm-wb/pull/1509
* https://github.com/kiegroup/kie-wb-distributions/pull/1118

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by akumar074** [comment]
Jenkins retest this

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by akumar074** [comment]
Jenkins retest this

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by akumar074** [comment]
Jenkins retest this

**Comment by karreiro** [comment]
Jenkins please do fdb.

**Comment by karreiro** [comment]
Jenkins retest this

**Comment by karreiro** [comment]
Jenkins retest this

**Comment by akumar074** [comment]
Jenkins retest this

**Review by karreiro — COMMENTED** [review]
Thank you, @akumar074! The DMN side seems great (I'm just double-checking it locally before the fine approach).

Also, I've left a minor question inline.

Thanks again for this PR!

**Review by tiagobento — APPROVED** [review]

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleResourcePaths.java [hunk 1]
```diff
@@ -24,5 +24,5 @@
 
     String PROJECT_REPOSITORIES_PATH = "project.repositories";
 
-    String PACKAGE_NAME_WHITE_LIST = "package-names-white-list";
+    String PACKAGE_NAME_ALLOW_LIST = "package-names-allow-list";
```
Comment (barboras7) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleResourcePaths.java:27]:
```suggestion
    String PACKAGE_NAMES_ALLOW_LIST = "package-names-allow-list";
```

**Review by romartin — APPROVED** [review]
Thanks @akumar074 , changes around BPMN editor are looking good. :+1:

**Review by karreiro — APPROVED** [review]
Thank you, @akumar074! I just finished my local check and the DMN editor is fine! Also, I'm approving because I don't consider the minor comment above a blocker :-)

Thanks again for this ensemble!

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.java [hunk 2]
```diff
@@ -747,10 +747,10 @@
     public static final String Ok = "Ok";
 
     @TranslationKey(defaultValue = "")
-    public static final String MasterIsRequired = "MasterIsRequired";
+    public static final String MainIsRequired = "MasterIsRequired";
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.java:750]: @karreiro You are right. I may have missed this. Based on @barboras7 comment below, I think we can use `DefaultBranch` in variable name like below. wdyt?
```suggestion
    public static final String DefaultBranchIsRequired = "MainIsRequired";
```

File: kie-wb-common-ala/kie-wb-common-ala-openshift/kie-wb-common-ala-openshift-provider/src/test/java/org/guvnor/ala/openshift/executor/OpenShiftExecutorTest.java [hunk 3]
```diff
@@ -200,14 +200,14 @@ public void testPipelineAndLifecycle() throws Exception {
...
+            put(GitConfig.BRANCH,  "main");
             put(GitConfig.CREATE_REPO,  "true");
             put(GitConfig.OUT_DIR,  workDir.getAbsolutePath());
             put(MavenProjectConfig.PROJECT_DIR, "decisionserver/hellorules");
             */
             // provider properties
-            put(KUBERNETES_MASTER.inputKey(),
-                "https://ce-os-rhel-master.usersys.redhat.com:8443");
+            put(KUBERNETES_MAIN.inputKey(),
+                "https://ce-os-rhel-main.usersys.redhat.com:8443");
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-ala/kie-wb-common-ala-openshift/kie-wb-common-ala-openshift-provider/src/test/java/org/guvnor/ala/openshift/executor/OpenShiftExecutorTest.java:210]: yes, I am changing the URL to include the main.

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ProjectImportServiceImplTest.java [hunk 4]
```diff
@@ -235,7 +235,7 @@ public void testGetProjects_DefaultDescription() {
         when(moduleService.getAllModules(Mockito.<Branch>any())).thenReturn(new HashSet<Module>() {{
             add(module);
         }});
-        doReturn(Collections.singletonList("master")).when(service).getBranches(Mockito.<org.uberfire.java.nio.file.Path>any(),
+        doReturn(Collections.singletonList("main")).when(service).getBranches(Mockito.<org.uberfire.java.nio.file.Path>any(),
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ProjectImportServiceImplTest.java:238]: As we want to remove all the occurrences of `master` wherever it's possible, we can mock the branch here to be `main`, and it does not affect our test.
^ @tiagobento

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.properties [hunk 5]
```diff
@@ -587,8 +587,8 @@ BranchManagementWritePermissionTooltip=Grants access for the user to modify all
 BranchManagementDeletePermissionTooltip=Grants access for the user to delete the branch.
 BranchManagementDeployPermissionTooltip=Grants access for the user to deploy the project in the branch.
 BranchesToBeImported=Branches to be imported
-MasterIsRequired=The master branch must be imported.
-AtLeastMasterIsRequired=At least the master branch must be imported.
+MainIsRequired=The master branch must be imported.
+AtLeastMainIsRequired=At least the master branch must be imported.
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.properties:591]: As I mentioned [here](https://github.com/kiegroup/appformer/pull/1178#discussion_r685787194), The master should be removed wherever possible. I agree that readability has an impact here. Using `DefaultBranch` can be an alternative.
```suggestion
AtLeastDefaultBranchIsRequired=At least the master branch must be imported.
```
wdyt?
^ @tiagobento

File: kie-wb-common-services/kie-wb-common-datamodel/kie-wb-common-datamodel-backend/src/test/resources/DataModelPackageAllowListTest3/pom.xml [hunk 6]
```diff
@@ -5,7 +5,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <groupId>org.kie.example</groupId>
-  <artifactId>ProjectDataModelPackageWhiteListTest4</artifactId>
+  <artifactId>ProjectDataModelPackageAllowListTest3</artifactId>
```
[the remark this answers is not in the document]
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-datamodel/kie-wb-common-datamodel-backend/src/test/resources/DataModelPackageAllowListTest3/pom.xml:8]: The `artifactId` is not properly assigned to the test projects. Even though it doesn't require to run these `DataModelPackageAllowListTest` projects, I just changed those ids to be proper in each test project.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleResourcePaths.java (hunk 1 above)
Comment (akumar074) [reply kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieModuleResourcePaths.java:27]: Yes, you are right this looks more suitable here. Thanks, I'll change it.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.properties (hunk 5 above)
[the remark this answers is not in the document]
Comment (tiagobento) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.properties:591]: Thanks @barboras7 pointing that out. I like @akumar074's suggestion, that's more neutral.

File: kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ProjectImportServiceImplTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tiagobento) [reply kie-wb-common-screens/kie-wb-common-examples-screen/kie-wb-common-examples-screen-backend/src/test/java/org/kie/workbench/common/screens/examples/backend/server/ProjectImportServiceImplTest.java:238]: @barboras7 No reason to keep using `master` where we _can_ replace by `main`. +1 for keeping using `main` on this test.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (karreiro) [reply kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/resources/i18n/LibraryConstants.java:750]: +1 @akumar074!

**Comment by akumar074** [comment]
Jenkins run fdb

**Comment by akumar074** [comment]
Jenkins retest this

**Review by barboras7 — APPROVED** [review]

**Pull request merged**